packages/gatsby-image/src/index.js

```diff
@@ -525,38 +538,27 @@ class Image extends React.Component {
 
           {/* Show the blurry base64 image. */}
           {image.base64 && (
-            <picture>
-              <source media={image.media} srcSet={image.base64} />
-              {imageVariants && generateBase64Sources(imageVariants)}
-              <Img src={image.base64} {...placeholderImageProps} />
-            </picture>
+            <Placeholder
+              src={image.base64}
+              spreadProps={placeholderImageProps}
+              imageVariants={imageVariants}
+              generateSources={generateBase64Sources}
+            />
           )}
 
           {/* Show the traced SVG image. */}
           {image.tracedSVG && (
-            <picture>
-              <source media={image.media} srcSet={image.tracedSVG} />
-              {imageVariants && generateTracedSVGSources(imageVariants)}
-              <Img src={image.tracedSVG} {...placeholderImageProps} />
-            </picture>
+            <Placeholder
+              src={image.tracedSVG}
+              spreadProps={placeholderImageProps}
+              imageVariants={imageVariants}
+              generateSources={generateTracedSVGSources}
+            />
           )}
 
           {/* Once the image is visible, start downloading the image */}
           {this.state.isVisible && (
             <picture>
-              {image.srcSetWebp && (
-                <source
-                  type={`image/webp`}
-                  media={image.media}
-                  srcSet={image.srcSetWebp}
-                  sizes={image.sizes}
-                />
-              )}
-              <source
-                media={image.media}
-                srcSet={image.srcSet}
-                sizes={image.sizes}
-              />
               {imageVariants && generateImageSources(imageVariants)}
```

Comment:
The pastebin has the logical guard removed. `imageVariants` always exists now(even if it contains only a single element).